src/modules/share/services/grade.service.ts

```diff
@@ -0,0 +1,99 @@
+import { prisma } from '@lib/prisma/prisma';
+
+import {
+  CreateGradeServiceProps,
+  GetGradeListServiceProps,
+  UpdateGradeInfoByIdServiceProps,
+} from '@share/types';
+
+export default class GradeService {
+  static create = async ({ name, code }: CreateGradeServiceProps) => {
+    const existingGrade = await prisma.grade.findUnique({
+      where: {
+        code: code,
+      },
+    });
+
+    if (existingGrade) {
+      throw new Error('existed');
+    }
+
+    const newGrade = await prisma.grade.create({
+      data: {
+        name: name,
+        code: code,
+      },
+    });
+
```

Comment:
Should olny admin can add grade? If yes, please add comment in case u forget in te future

Reply:
it's will be done in the middleware tasks